src/Backups/BackupSettings.h

```diff
@@ -90,6 +90,13 @@ struct BackupSettings
     /// Is it allowed to use blob paths to calculate checksums of backup entries?
     bool allow_checksums_from_remote_paths = true;
 
+    /// Generator for a unique and deterministic backup file name.
+    /// If specified, should be one of ["first_file_name", "checksum"]. Defaults to "first_file_name".
```

Comment:
Please add some details here about these generators in the comment.

Reply:
What is the default value? Maybe it's better to write it here?
```
String data_file_name_generator = "first_file_name";
```

The same about the prefix length.

Reply:
Why is `data_file_name_generator` a String? Let's use enum.

Reply:
Used [SettingsEnums.cpp](https://github.com/ClickHouse/ClickHouse/pull/88418/files#diff-7c1f41a9953b16d6ec88ecf82563c5e488123a0bceb0bb24e03b6efd2b7c3e9a)